### PR TITLE
Balder/allow cased uncased

### PR DESCRIPTION
**Pull request by baldersheim**
@bratseth and @geirst PR
This only propagates correct attributes config for the backend.
Lowercasing control is up next.

### CODE REVIEW

**Review by bratseth — APPROVED** [review]
👍

File: config-model/src/main/java/com/yahoo/searchdefinition/document/Case.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.yahoo.searchdefinition.document;
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/searchdefinition/document/Case.java:1]: Please add copyright.

File: config-model/src/main/java/com/yahoo/searchdefinition/document/Case.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+public enum Case {
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/searchdefinition/document/Case.java:3]: Consider adding comment.

File: config-model/src/main/java/com/yahoo/searchdefinition/document/SDField.java [hunk 3]
```diff
@@ -554,6 +557,17 @@ public void setMatchingType(Matching.Type type) {
         }
     }
 
+    /**
+     * Set the matching type for this field and all subfields.
+     */
+    // TODO: When this is not the same as getMatching().setthis we have a potential for inconsistency. Find the right
```
Comment (geirst) [inline config-model/src/main/java/com/yahoo/searchdefinition/document/SDField.java:563]: Typo: 'setthis'

**Review by geirst — COMMENTED** [review]
👍

File: config-model/src/main/java/com/yahoo/searchdefinition/document/Case.java (hunk 1 above)
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/searchdefinition/document/Case.java:1]: Fixed

File: config-model/src/main/java/com/yahoo/searchdefinition/document/Case.java (hunk 2 above)
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/searchdefinition/document/Case.java:3]: Fixed

File: config-model/src/main/java/com/yahoo/searchdefinition/document/SDField.java (hunk 3 above)
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/searchdefinition/document/SDField.java:563]: I just copied the comment from setMatchingType and setMatchingAlgorithm just to ensure that they all got clenaed up when the original author of the comment decided to clean this up. Not sure what he really meant. Hope it was not me......

**Pull request merged**